src/impl.cpp

```diff
@@ -1 +1,134 @@
 #include "impl.h"
+
+LinkedList::LinkedList(): _first(nullptr), size(0) {}
+
+LinkedList::LinkedList(std::initializer_list<int> list): _first(nullptr), size(0)
+{
+    for (auto &it: list)
+    {
+        insertAt(size, it);
+    }
+}
+
+LinkedList::~LinkedList()
+{
+    Node *prev = _first;
+    Node *next;
+
+    while(prev)
+    {
+        next = prev->next;
+        delete prev;
+        prev = next;
+    }
+}
+
+void LinkedList::removeAt(int index)
+{
+    int counter = 0;
+    if (!getLength() || index >= getLength()) return;
+
+    Node *removeNode;
+
+    if (index == 0)
+    {
+        removeNode = _first;
+        _first = _first->next;
+        delete removeNode;
+    }
+    else
+    {
+        Node *currNode = _first;
+        Node *prevNode;
+        while(counter < index && (prevNode = currNode) && (currNode = currNode->next))
+        {
+            counter++;
+        }
+
+        if (index == counter)
+        {
+            removeNode = currNode;
+            prevNode->next = currNode->next;
+            delete removeNode;
+        }
+    }
+    size--;
+}
+
+void LinkedList::insertAt(int index,int value)
+{
+    Node *prev = _first;
+    Node *next;
+    int nextPosition = 0;
+
+    if (index <= size)
```

Comment:
please think how it could be implemented easier
or extract sub-blocks to the named private methods